doc/manual/modules.rst

```diff
@@ -236,7 +236,29 @@ Namespace miscellanea
 ---------------------
 
 If a name is qualified (e.g. ``Base.sin``), then it can be accessed even if
-it is not exported. This is often useful when debugging.
+it is not exported. This is often useful when debugging. Normally Julia
+variables, functions, types, and the fields of composite types are publicly
+accessible. Julia does not have corresponding "private" versions of these
+items that are inaccessible from outside the module. However, a common
+Julia convention is that names prefixed with an underscore (e.g.,
+``_foo()``) are private implementation details.
+
+Exported names form the public facing API of a module. In some cases,
+unexported names are also part of the public API, such as found in the
+``Pkg`` and ``Profile`` modules in Base. In these cases, the unexported
+names should be clearly documented. Changes to public APIs should be made
+with care. Registered packages should bump their version number
+appropriately to signal the breaking change. Deprecation warnings are also
+appropriate.
+
+Unexported names are not strictly part of the public API and may change at
```

Comment:
>I should add that it would be nice to have a formal way to declare that something like JSON.parse(string) is part of the interface, outside of just documentation. That was discussed briefly in the other issue.

That's just what I've been trying to get people to agree to, with `public JSON.parse` :grinning:
because it really is orthogonal to whether or not something has documentation.